### PR TITLE
Improve README

### PR DESCRIPTION
**Pull request by resolritter**
You can read this new README at https://github.com/resolritter/nelua-lang/blob/patch-1/README.md

# Rationality
  - The site should be the source of truth for documentation; the README should only reference it to avoid duplication of information
  - The README should not have technical or in-depth documentation upfront, but instead quick information and links to other places
      - Newcomers looking for a new language want to be able to read the examples and the syntax ASAP
      - Experienced programmers want to see the code and what the language can do ASAP
      - Sections which relate to those interests should be merged

# Quick installation
  - The installation is already linked in the Tutorial, so it doesn't need to be written in the README, as it can get outdated and/or misleading with time

# About, Roadmap, etc

- The description here should be included in the official website as well. Maybe the website code could generate the HTML from this README markdown.

### CODE REVIEW

**Comment by edubart** [comment]
I agree with some stuff, the readme should be smaller now, since I've included more information in the site, and will be less for me to maintain and update.

But some changes I don't feel comfortable, I will comment on each one.

File: README.md [hunk 1]
```diff
@@ -2,74 +2,145 @@
...
+for performance-sensitive applications where [Lua](https://en.wikipedia.org/wiki/Lua_(programming_language))
+would not be efficient, such as operating systems, real-time applications and
+game engines. While it has syntax and semantics similar to Lua, it primarily
+focuses on generating efficient C code and provide support for
+highly-optimizable low-level programming. Using Nelua idioms such as records,
+arrays, manual memory management and pointers should result in performance as
+efficient as pure C; on the other hand, when using Lua idioms such as tables,
+metatables and untyped variables, the compiler will bake a runtime library for
+this sort of dynamic functionality into the program, which could incur some
+noticeable runtime cost.
```
Comment (edubart) [inline README.md:100]: I like the changes heres, except the `noticeable runtime cost` could be replaced with `some runtime overhead`, not all dynamic functionally incur noticeable runtime cost, some are very minor.

File: README.md [hunk 2]
```diff
@@ -2,74 +2,145 @@
...
-For example in Lua classes don't exist but you can implement yourself using metatables,
-in Nelua they don't exist neither but you can implement more efficiently at compile time
-by meta programming or at runtime just like in Lua.
+[polymorphism](https://en.wikipedia.org/wiki/Polymorphism_(computer_science)),
+all without having to add them into the core specification, thus keeping the
+language simple, extensible and compact. The same way that Lua's
+object-oriented patterns are not built into the language, but can be
+nonetheless achieved through [metatables](https://webserver2.tecgraf.puc-rio.br/lua/local/pil/13.html),
+in Nelua you could yourself implement a similar functionality which is fully
+decided at compile time or dynamically dispatched at runtime.
```
Comment (edubart) [inline README.md:117]: Good changes here.

File: README.md [hunk 3]
```diff
@@ -2,74 +2,145 @@
...
+be created. Another benefit is that Nelua can reach the same target platforms
+as C99 compilers, such [GCC](https://en.wikipedia.org/wiki/GNU_Compiler_Collection)
+or [Clang](https://en.wikipedia.org/wiki/Clang), while also enjoying
+state-of-the-art compiler optimizations provided by them.
+
+The initial motivation for its creation was to replace C/C++ parts of projects
+which currently uses Lua with a language that has syntax and semantics similar
+to Lua, but allows for fine-grained performance optimizations and does not lose
+the ability to go low level, therefore unifying the syntax and semantics across
+both compiled and dynamic languages.
```
Comment (edubart) [inline README.md:143]: Nice changes here too!

File: README.md [hunk 4]
```diff
@@ -2,74 +2,145 @@
...
 
-Nelua is a minimal, simple, efficient, statically typed, compiled, meta programmable,
-safe and extensible systems programming language with a Lua flavor.
-It uses ahead of time compilation to compile to native code.
-Nelua stands for *Native Extensible Lua*.
+Nelua (stands for **N**ative **E**xtensible **Lua**) is a minimal,
+statically-typed and meta-programmable systems programming language heavily
+inspired by Lua. C99 is its main compilation target, but it can also compile to
+Lua (do note that the Lua backend is not as feature-complete, nor has as much
+support at this time).
```
[the remark this answers is not in the document]
Comment (resolritter) [reply README.md]: I've removed the Lua backend part and removed the notice about C being the "main compilation target", now it just says that it compiles to C which is a relevant selling point to bring upfront IMO.

File: README.md [hunk 5]
```diff
@@ -2,74 +2,145 @@
...
+
+The language is in alpha state and still evolving. Visit the
+[Roadmap](https://github.com/edubart/nelua-lang/projects) to know what's
+actively being developed or read the [Draft](https://nelua.io/draft/) to have a
+rough idea of what could be included in the future.
```
[the remark this answers is not in the document]
Comment (resolritter) [reply README.md]: Removed most of this part

File: README.md [hunk 6]
```diff
@@ -2,74 +2,145 @@
...
+
+After installing, you might want to check out the featured example: a Snake
+game leveraging the famous [SDL2](https://www.libsdl.org) game engine.
+
+``` nelua examples/snakesdl.nelua ```
+
+Last, but not least, feel free to give feedback and ask questions over at the
+[Discord server](https://discord.gg/7aaGeG7).
+
+## Community showcase
```
[the remark this answers is not in the document]
Comment (resolritter) [reply README.md]: This part was removed. I don't have access to the wiki, but you can paste this Markdown there and link it in the README after this gets merged.

```
## Community showcase

### Games

The following games are all playable in the browser (built through Emscripten).

- [Tetris game clone](https://edubart.github.io/nelua-tetris/) ([source](https://github.com/edubart/nelua-tetris))
- [2048](https://edubart.github.io/nelua-game2048/) ([source](https://github.com/edubart/nelua-game2048))
- [Celeste game clone](https://edubart.github.io/neco/celeste.html)

### Game-related

- [Raytracer](https://github.com/edubart/lunray)
- [Game profiler](https://github.com/Andre-LA/nprof)
- [Raylib bindings](https://github.com/Andre-LA/raylib-nelua-mirror)

![demo](https://media.discordapp.net/attachments/680417079542939713/740021587285245992/Peek_2020-08-03_22-35.gif)
```

File: README.md (hunk 1 above)
Comment (resolritter) [reply README.md]: Changed to "some runtime overhead"

File: README.md (hunk 6 above)
[the remark this answers is not in the document]
Comment (edubart) [reply README.md]: Oh I though the wiki was editable by anyone, should have been, will look into that.

[the remark this answers is not in the document]
Comment (edubart) [reply README.md]: It should be editable now for anyone I think, please confirm for me if you can edit.

**Comment by edubart** [comment]
Approved, will merge soon.

**Comment by edubart** [comment]
Merged in https://github.com/edubart/nelua-lang/commit/ada91796eb26561d4522f715e3ef60f13d6d7bfb

**Pull request closed without merging**